controllers/componentdetectionquery_controller_conditions.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2022 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	appstudiov1alpha1 "github.com/redhat-appstudio/application-service/api/v1alpha1"
+)
+
+func (r *ComponentDetectionQueryReconciler) SetCreateConditionAndUpdateCR(ctx context.Context, componentDetectionQuery *appstudiov1alpha1.ComponentDetectionQuery, createError error) {
```

Comment:
So I am wondering if in the future we can update it?

For instance, someone goes thru the import flow and provides a repo, HAC submits a query detection CR and sees there is no devfile. User then updates the repo; would HAC create a new CR? Probably yes at this point.

But that would also mean, HAC would need to delete these CRs? Otherwise there would be like old CRs lying around 🤔

Reply:
updated to `Completed`